src/random_test.cpp

```diff
@@ -1607,20 +1608,64 @@ void Table::DeleteRandomRow(Thd1 *thd) {
       }
     }
   }
-
   std::string sql = "DELETE FROM " + name_ + " WHERE " +
-                    columns_->at(where)->name_ + "=" +
-                    columns_->at(where)->rand_value();
+                   columns_->at(where)->name_;
+  if (prob <= 90)
+    sql += " = " + columns_->at(where)->rand_value();
+  else if (prob <= 92)
+    sql += " >= " +
+          columns_->at(where)->rand_value() + " AND " +
+          columns_->at(where)->name_ + " <= " +
+          columns_->at(where)->rand_value();
+  else if (prob <= 94)
+    sql += " IN (" +
+          columns_->at(where)->rand_value() + "," +
+          columns_->at(where)->rand_value() + ")";
+  else if (prob <= 96)
+    sql += " BETWEEN " +
+          columns_->at(where)->rand_value() + " AND " +
+          columns_->at(where)->rand_value();
+  else
+    sql += " LIKE '%" +
+          columns_->at(where)->rand_value() + "%'";
+
   table_mutex.unlock();
   execute_sql(sql, thd);
 }
 
 void Table::SelectRandomRow(Thd1 *thd) {
   table_mutex.lock();
   auto where = rand_int(columns_->size() - 1);
+  auto prob = rand_int(102);
   std::string sql = "SELECT * FROM " + name_ + " WHERE " +
-                    columns_->at(where)->name_ + "=" +
-                    columns_->at(where)->rand_value();
+                   columns_->at(where)->name_;
+  if (prob <= 90)
+    sql += " = " +
+          columns_->at(where)->rand_value();
+  else if (prob <= 92)
+    sql += " >= " +
+          columns_->at(where)->rand_value();
+  else if (prob <= 94)
+    sql += " >= " +
+          columns_->at(where)->rand_value() + " AND " +
+          columns_->at(where)->name_ + " <= " +
+          columns_->at(where)->rand_value();
+  else if (prob <= 96)
+    sql += " IN (" +
+          columns_->at(where)->rand_value() + ", " +
+          columns_->at(where)->rand_value() + ")";
+  else if (prob <= 98)
+    sql += " LIKE '%" +
+          columns_->at(where)->rand_value() + "%'";
+  else if (prob <= 100)
+    sql += " BETWEEN " +
+          columns_->at(where)->rand_value() + " AND " +
+          columns_->at(where)->rand_value();
+  else if (rand_int(500) == 0)
```

Comment:
if prob =101,102 and rand(500) is not 0 then it will not execute any sql.
reorder them so it will at lease execute some sql.

Reply:
Fixed